store/search/ts.go

```diff
@@ -38,72 +40,88 @@ type storeImplWithMetrics struct {
 	s Store
 }
 
+func (m *storeImplWithMetrics) measure(ctx context.Context, name string, f func(ctx context.Context) error) {
+	// Low level measurement wrapper that is called by the measure functions on the appropriate receiver
+	tags := metrics.GetSearchTags(ctx, name)
+	if config.DefaultConfig.Metrics.Search.ResponseTime {
+		metrics.SearchRequests.Tagged(tags).Histogram("histogram", tally.DefaultBuckets)
+		defer metrics.SearchRequests.Tagged(tags).Histogram("histogram", tally.DefaultBuckets).Start().Stop()
+	}
+	if config.DefaultConfig.DatadogTrace.Enabled {
+		parentSpan, exists := tracer.SpanFromContext(ctx)
+		if exists {
+			spanOpts := []tracer.StartSpanOption{
+				tracer.ServiceName("tigris.search"),
+				tracer.ResourceName(name),
+				tracer.SpanType("search"),
+				tracer.Measured(),
+			}
+			// There is a parent span, keep on tracing in the kv layer
+			spanOpts = append(spanOpts, tracer.ChildOf(parentSpan.Context()))
+			span := tracer.StartSpan(name, spanOpts...)
+			defer span.Finish()
+			ctx = tracer.ContextWithSpan(ctx, span)
+		}
+	}
+	err := f(ctx)
+	if err == nil {
```

Comment:
We have a nice way to do the log tracing here instead of higher above as the Span ID is exposed. I'd just add a // TODO for now to remind me later and follow up with a separate PR to address that.

Reply:
I will add the log related things in the next PR.